Fix struct layout for FBIOGET_FSCREENINFO ioctl.

The structure uses plain __u64 for the physical addresses
in the kernel and does not need capabilities. Changing these
to pointers causes the structure layout to change on CHERI
systems.

Upstream-Status: Inappropriate [CHERI specific]

diff --git a/hw/xfree86/fbdevhw/fbpriv.h b/hw/xfree86/fbdevhw/fbpriv.h
index 8d224ae..8b1b154 100644
--- a/hw/xfree86/fbdevhw/fbpriv.h
+++ b/hw/xfree86/fbdevhw/fbpriv.h
@@ -91,7 +91,7 @@
 
 struct fb_fix_screeninfo {
     char id[16];                /* identification string eg "TT Builtin" */
-    char *smem_start;           /* Start of frame buffer mem */
+    unsigned long smem_start;   /* Start of frame buffer mem */
     /* (physical address) */
     __u32 smem_len;             /* Length of frame buffer mem */
     __u32 type;                 /* see FB_TYPE_*                */
@@ -101,7 +101,7 @@ struct fb_fix_screeninfo {
     __u16 ypanstep;             /* zero if no hardware panning  */
     __u16 ywrapstep;            /* zero if no hardware ywrap    */
     __u32 line_length;          /* length of a line in bytes    */
-    char *mmio_start;           /* Start of Memory Mapped I/O   */
+    unsigned long mmio_start;           /* Start of Memory Mapped I/O   */
     /* (physical address) */
     __u32 mmio_len;             /* Length of Memory Mapped I/O  */
     __u32 accel;                /* Type of acceleration available */
